repl/tests/rexpect.rs

```diff
@@ -22,126 +19,137 @@ macro_rules! test {
     };
 }
 
+struct REPL {
+    session: PtySession,
+    prompt: &'static str,
+}
+
+impl REPL {
+    /// Defines the command, timeout, and prompt settings.
+    /// Wraps a rexpect::session::PtySession. expecting the prompt after launch.
+    fn new() -> Result<REPL> {
+        let command = "../target/debug/gluon -i";
+        let timeout: u64 = 10_000;
+        let prompt: &'static str = "> ";
+
+        let mut session = spawn(command, Some(timeout))?;
+        session.exp_string(prompt)?;
+
+        Ok(REPL { session, prompt })
+    }
+
+    /// Ensures certain lines are expected to reduce race conditions.
+    /// If no ouput is expected or desired to be tested, pass it an Option::None,
+    /// causing rexpect to wait for the next prompt.
+    fn test(&mut self, send: &str, expect: Option<&str>) -> Result<()> {
```

Comment:
Since we wrap `rexpect` I think we might as well do `unwrap_or_else` in these wrapper methods. That would let the `test` macro be removed (making it simpler to add more tests later).

```rust
fn test(&mut self, send: &str, expect: Option<&str>) {
    self.test_(send, expect).unwrap_or_else(|err| panic!("{}", err));
}
fn test_(&mut self, send: &str, expect: Option<&str>) -> Result<()> {
```